lib/database.js

```diff
@@ -133,63 +136,63 @@ module.exports = class Database extends EventEmitter {
       return Promise.resolve(this.connection);
     }
 
-    if (typeof this.connectionString == 'string') {
-          return mongodb.MongoClient
-            .connect(this.connectionString, Object.assign(this.options))
-            .then(client => {
+    if (this._connecting) {
+      // Race condition, return the promise
+      return this._connecting;
+    }
+
```

Comment:
nit: this code flow is difficult to read since it uses an IIFE. Can it be refactored?

Hint - 
<img width="638" alt="image" src="https://github.com/user-attachments/assets/05b02bad-2bdd-46fc-b6b9-ba3c6860b0f4">